cli/generate-test-fixtures.js

```diff
@@ -0,0 +1,144 @@
+let featuresUsed = {}; // feature id -> times used
+for (const featureFile of fs.readdirSync(fixFeaturesDir)) {
+  if (path.extname(featureFile) === '.js') {
+    let fixFeature = require(path.join(fixFeaturesDir, featureFile));
+    fixFeature.id = path.basename(featureFile, '.js');
+    if (!('order' in fixFeature)) {
+      fixFeature.order = 0;
+    }
+    fixFeatures.push(fixFeature);
+    featuresUsed[fixFeature.id] = 0;
+  }
+}
+fixFeatures.sort((a, b) => {
+  if (a.order === b.order) {
+    if (a.name.toLowerCase() > b.name.toLowerCase()) {
+      return 1;
+    }
+    else if (a.name.toLowerCase() < b.name.toLowerCase()) {
+      return -1;
+    }
+  }
+  return b.order - a.order;
+});
+
+
+// check which features each fixture supports
+let fixtures = [];
+const manufacturers = JSON.parse(fs.readFileSync(path.join(fixturesDir, 'register.json'), 'utf8')).manufacturers;
+for (const man of Object.keys(manufacturers)) {
+  for (const fix of manufacturers[man]) {
+    // pre-process data
+    const fixData = JSON.parse(fs.readFileSync(path.join(fixturesDir, man, fix + '.json'), 'utf8'));
+    let fineChannels = {};
+    for (const ch of Object.keys(fixData.availableChannels)) {
+      const channel = fixData.availableChannels[ch];
+
+      if ('fineChannelAliases' in channel) {
+        for (const alias of channel.fineChannelAliases) {
+          fineChannels[alias] = ch;
+        }
+      }
+    }
+
+    let fixture = {
+      man: man,
+      key: fix,
+      name: fixData.name,
+      features: []
+    };
+    fixtures.push(fixture);
+
+    // check all features
+    for (const fixFeature of fixFeatures) {
+      if (fixFeature.hasFeature(fixData, fineChannels)) {
+        fixture.features.push(fixFeature.id);
+        featuresUsed[fixFeature.id]++;
+      }
+    }
+  }
+}
+
+// compress list
+if (!args.all) {
+  // sort out fixtures with least features first
+  fixtures.sort((a, b) => {
+    return a.features.length - b.features.length;
+  });
+  // sort out
+  fixtures = fixtures.filter(fixture => {
+    for (const feature of fixture.features) {
+      if (featuresUsed[feature] === 1) {
+        return true;
+      }
+    }
+    // has no new features, so remove it
+    for (const feature of fixture.features) {
+      featuresUsed[feature]--;
+    }
+    return false;
+  });
+  // original ordering
+  fixtures.sort((a, b) => {
+    const manFixA = `${a.man}/${a.key}`;
+    const manFixB = `${b.man}/${b.key}`;
+    return manFixA > manFixB ? 1 : manFixA < manFixB ? -1 : 0;
+  });
+
+  console.log(colors.yellow('Generated list of test fixtures:'));
+  for (const fixture of fixtures) {
+    console.log(` - ${fixture.man}/${fixture.key}`);
+  }
+  let testFixturesFile = path.join(__dirname, '..', 'tests', 'test-fixtures.json');
+  fs.writeFileSync(testFixturesFile, JSON.stringify(fixtures, null, 2));
+  console.log(`\nSuccessfully updated ${testFixturesFile}.\n`);
+}
+
+// generate markdown code
+const markdown = []; // lines
```

Comment:
`let` instead of `const`. Also, why not name it `mdLines` and remove the comment?